### PR TITLE
Finalize Integrations

### PR DESCRIPTION
**Pull request by snaselj**
# Closes NaN

Finish adding integrations.

## Done

- Added Cisco ACI documentation.
- Added conflicting apps assertion.

## To-Do

- [x] Add Arista CloudVision integration documentation.
- [x] Add Infoblox integration documentation.
- [x] Add IPFabric integration documentation.
- [x] Add ServiceNow integration documentation.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -22,6 +22,19 @@ The nautobot-ssot plugin builds atop the [DiffSync](https://github.com/networkto
 * The outcome of executing of a data synchronization Job is automatically saved to Nautobot's database for later review.
 * Detailed logging output generated by DiffSync is automatically captured and saved to the database as well.
 
+### Integrations
+
+This plugin supports following integrations:
```
Comment (cmsirbu) [inline README.md:27]:
```suggestion
This plugin supports the following integrations:
```

File: README.md [hunk 2]
```diff
@@ -22,6 +22,19 @@ The nautobot-ssot plugin builds atop the [DiffSync](https://github.com/networkto
...
+
+- Cisco ACI
+- Arista CloudVision
+- Infoblox
+- IPFabric
+- ServiceNow
+
+Read more about integrations [here](https://docs.nautobot.com/projects/ssot/en/latest/user/integrations).
+
+To install integrations follow [this guide](https://docs.nautobot.com/projects/ssot/en/latest/admin/install/#integrations-configuration).
```
Comment (cmsirbu) [inline README.md:37]:
```suggestion
Read more about integrations [here](https://docs.nautobot.com/projects/ssot/en/latest/user/integrations). To enable and configure integrations follow the instructions from [the install guide](https://docs.nautobot.com/projects/ssot/en/latest/admin/install/#integrations-configuration).
```

File: docs/admin/install.md [hunk 3]
```diff
@@ -8,6 +8,30 @@
 !!! note
     Please check the [dedicated page](compatibility_matrix.md) for a full compatibility matrix and the deprecation policy.
 
+### Potential Apps Conflicts
+
+!!! warning
```
Comment (cmsirbu) [inline docs/admin/install.md:13]: Leave this warning in the prerequisites section and link to the upgrade page. It'll be good for visibility. 

As all the instructions in this subsection are for upgrading, it should move to the upgrade guide.

File: docs/admin/install.md [hunk 4]
```diff
@@ -65,3 +89,14 @@ The plugin behavior can be controlled with the following list of settings:
 | ------------------- | ------- | ------- | ---------------------------------------------------------- |
 | `hide_example_jobs` | `True`  | `False` | A boolean to represent whether or display the example job. |
 
+## Integrations Configuration
+
+The `nautobot-ssot` package includes multiple integrations. Each requires extra dependencies defined in `pyproject.toml`. See the [installation guide](#installation-guide) for instructions on installing these dependencies.
```
Comment (cmsirbu) [inline docs/admin/install.md:94]: This link seems wrong and is the dependencies info missing?

File: docs/admin/install.md [hunk 5]
```diff
@@ -65,3 +89,14 @@ The plugin behavior can be controlled with the following list of settings:
...
+
+Set up integrations using the specific guides:
```
Comment (cmsirbu) [inline docs/admin/install.md:96]:
```suggestion
Set up each integration using the specific guides:
```

File: docs/user/integrations.md [hunk 6]
```diff
@@ -0,0 +1,9 @@
+# Integrations
+
+This plugin supports following integrations:
```
Comment (cmsirbu) [inline docs/user/integrations.md:3]:
```suggestion
This plugin supports the following integrations:
```

File: docs/user/integrations.md
Comment (cmsirbu) [inline docs/user/integrations.md:1]: Let's move this to an "integrations" folder and rename it to index.md and move the other pages inside as well. See mkdocs.yml comment.

File: mkdocs.yml [hunk 7]
```diff
@@ -101,6 +101,7 @@ nav:
       - App Overview: "user/app_overview.md"
       - Getting Started: "user/app_getting_started.md"
       - Using the App: "user/app_use_cases.md"
+      - Integrations: "user/integrations.md"
```
Comment (cmsirbu) [inline mkdocs.yml:104]:
```suggestion
      - Integrations: 
           - "user/integrations/index.md"
           - Cisco ACI: "user/integrations/aci.md"
```

As per other comment to structure folders.

File: docs/user/aci.md [hunk 8]
```diff
@@ -0,0 +1,66 @@
+# Cisco ACI SSoT Integration
+
+Cisco ACI SSoT integration is built as part of the [Nautobot Single Source of Truth (SSoT)](https://github.com/nautobot/nautobot-plugin-ssot) App. SSoT App enables Nautobot to be the aggregation point for data coming from multiple systems of record (SoR).
```
Comment (cmsirbu) [inline docs/user/aci.md:3]:
```suggestion
The Cisco ACI SSoT integration is built as part of the [Nautobot Single Source of Truth (SSoT)](https://github.com/nautobot/nautobot-plugin-ssot) App. The SSoT App enables Nautobot to be the aggregation point for data coming from multiple systems of record (SoR).
```

File: docs/user/aci.md [hunk 9]
```diff
@@ -0,0 +1,66 @@
...
+
+You use the plugin by navigating to **Plugins > Dashboard** in Nautobot.  Then click on **Cisco ACI Data Source**.
+
+![Nautobot SSoT Dashboard](../images/aci-nautobot-ssot-dashboard.png)
+ 
+From the **Cisco ACI Data Source** page you can click **Sync Now** to begin a synchronization job and view the history of synchronization jobs.
+
+![SSoT ACI Dashboard](../images/aci-dashboard.png)
+
+After clicking **Sync Now**, you can select whether you would like to do a dry-run as well as schedule when you would like the job to run.  With a dry-run, you can see what information will be brought from ACI into Nautobot without actually performing the synchronization. The job can be run immediately, scheduled to run at a later date/time, or configured to run hourly, daily, or weekly at a specified date/time.
```
Comment (cmsirbu) [inline docs/user/aci.md:42]:
```suggestion
After clicking **Sync Now**, you can select whether you would like to do a dry-run as well as schedule when you would like the job to run. With a dry-run, you can see what information will be brought from ACI into Nautobot without actually performing the synchronization. The job can be run immediately, scheduled to run at a later date/time, or configured to run hourly, daily, or weekly at a specified date/time. 
```

File: docs/user/aci.md [hunk 10]
```diff
@@ -0,0 +1,66 @@
+# Cisco ACI SSoT Integration
+
+Cisco ACI SSoT integration is built as part of the [Nautobot Single Source of Truth (SSoT)](https://github.com/nautobot/nautobot-plugin-ssot) App. SSoT App enables Nautobot to be the aggregation point for data coming from multiple systems of record (SoR).
+
+To accomplish the synchronization of data, the SSoT ACI integrations communicates with the Cisco ACI controller, the Application Policy Infrastructure Controller (APIC). The APIC provides a central point of administration for the ACI fabric via a web dashboard or REST API.
+
+The SSoT ACI integration eliminates the need for manually adding objects to Nautobot that have been automatically discovered by the Cisco APIC controller.  This includes information such as device model/serial numbers, node management IP addressing, and more.
```
Comment (cmsirbu) [inline docs/user/aci.md:7]:
```suggestion
The SSoT ACI integration eliminates the need for manually adding objects to Nautobot that have been automatically discovered by the Cisco APIC controller. This includes information such as device model/serial numbers, node management IP addressing, and more.
```

**Review by cmsirbu — COMMENTED** [review]
Looks good! Don't forget to also add the contributor thanks in the readme for all the individual repos the integrations came from.

File: docs/admin/install.md (hunk 3 above)
Comment (snaselj) [reply docs/admin/install.md]: Fixed

File: docs/admin/install.md (hunk 4 above)
Comment (snaselj) [reply docs/admin/install.md]: Fixed

File: docs/user/integrations.md
Comment (snaselj) [reply docs/user/integrations.md:1]: Fixed

**Comment by snaselj** [comment]
> Looks good! Don't forget to also add the contributor thanks in the readme for all the individual repos the integrations came from.

All comments are addressed now, could you pls. check it @cmsirbu ?

File: README.md [hunk 11]
```diff
@@ -83,3 +81,14 @@ Any PRs with fixes or improvements are very welcome!
...
+## Acknowledgements
+
+This project includes code originally written in separate plugins, which have been merged into this project:
+
+- [nautobot-plugin-ssot-aci](https://github.com/nautobot/nautobot-plugin-ssot-aci):
+    Thanks
+    [@chadell](https://github.com/chadell),
+    [@dnewood](https://github.com/dnewood),
+    [@progala](https://github.com/progala),
+    [@ubajze](https://github.com/ubajze),
```
Comment (cmsirbu) [inline README.md:94]:
```suggestion
    [@ubajze](https://github.com/ubajze)
```

File: docs/admin/install.md [hunk 12]
```diff
@@ -43,6 +22,12 @@ The plugin is available as a Python package via PyPI and can be installed with `
 pip install nautobot-ssot
 ```
 
+To use specific integrations, add them as extra dependencies:
+
+```shell
+pip install nautobot-ssot[aci]
+```
+
 To ensure Single Source of Truth is automatically re-installed during future upgrades, create a file named `local_requirements.txt` (if not already existing) in the Nautobot root directory (alongside `requirements.txt`) and list the `nautobot-ssot` package:
```
Comment (cmsirbu) [inline docs/admin/install.md:31]:
```suggestion
To ensure Single Source of Truth is automatically re-installed during future upgrades, create a file named `local_requirements.txt` (if not already existing) in the Nautobot root directory (alongside `requirements.txt`) and list the `nautobot-ssot` package and any of the extras:
```

File: docs/admin/install.md [hunk 13]
```diff
@@ -91,12 +76,12 @@ The plugin behavior can be controlled with the following list of settings:
 
 ## Integrations Configuration
 
-The `nautobot-ssot` package includes multiple integrations. Each requires extra dependencies defined in `pyproject.toml`. See the [installation guide](#installation-guide) for instructions on installing these dependencies.
+The `nautobot-ssot` package includes multiple integrations. Each requires extra dependencies defined in `pyproject.toml`. See the [install guide](#install-guide) for instructions on installing these dependencies.
```
Comment (cmsirbu) [inline docs/admin/install.md:79]:
```suggestion
The `nautobot-ssot` package includes multiple integrations. Each requires extra dependencies defined in `pyproject.toml`.
```

Comment (cmsirbu) [reply docs/admin/install.md:79]: No need to distract by sending them to the start again, each integration setup page reinforces the extras that need to be installed.

File: docs/admin/integrations/aci_setup.md [hunk 14]
```diff
@@ -87,14 +87,14 @@ The YAML files need to be placed in the `nautobot_ssot/integrations/aci/diffsync
...
-![APIC Fabric Dashboard](../images/aci-apic-fabric-dashboard.png)
+![APIC Fabric Dashboard](../../images/aci-apic-fabric-dashboard.png)
 
 There are example YAML files for a few common switch models in `nautobot_ssot/integrations/aci/diffsync/device-types`, and additional ones can be downloaded [here](https://github.com/netbox-community/devicetype-library/tree/master/device-types/Cisco).
 
 ## Upgrading from `nautobot-plugin-ssot-aci` App
 
 !!! warning
-    When upgrading from `nautobot-plugin-ssot-aci` App, it's necessary to [avoid conflicts](install.md#potential-apps-conflicts).
+    When upgrading from `nautobot-plugin-ssot-aci` App, it's necessary to [avoid conflicts](../install.md#potential-apps-conflicts).
```
Comment (cmsirbu) [inline docs/admin/integrations/aci_setup.md:97]:
```suggestion
    When upgrading from `nautobot-plugin-ssot-aci` App, it's necessary to [avoid conflicts](../upgrade.md#potential-apps-conflicts).
```

File: docs/admin/upgrade.md [hunk 15]
```diff
@@ -3,3 +3,27 @@
...
+- `nautobot_plugin_ssot_aci`
+- `nautobot_plugin_ssot_arista_cloudvision`
+- `nautobot_plugin_ssot_infoblox`
+- `nautobot_plugin_ssot_ipfabric`
+- `nautobot_plugin_ssot_servicenow`
+
+To prevent conflicts during `nautobot-ssot` upgrade:
+
+- Remove conflicting applications from the `PLUGINS` section in your Nautobot configuration before enabling the latest `nautobot-ssot` version.
+- Transfer the configuration for conflicting apps to the `PLUGIN_CONFIG["nautobot_ssot"]` section of your Nautobot configuration. See `development/nautobot_config.py` for an example. Each [integration set up guide](#integrations-configuration) contains a chapter with upgrade instructions.
```
Comment (cmsirbu) [inline docs/admin/upgrade.md:23]:
```suggestion
- Transfer the configuration for conflicting apps to the `PLUGIN_CONFIG["nautobot_ssot"]` section of your Nautobot configuration. See `development/nautobot_config.py` for an example. Each [integration set up guide](../integrations/) contains a chapter with upgrade instructions.
```

**Comment by snaselj** [comment]
@cmsirbu

Added Arista CloudVision integration, could you pls. check it?

File: README.md [hunk 16]
```diff
@@ -22,6 +22,17 @@ The nautobot-ssot plugin builds atop the [DiffSync](https://github.com/networkto
 * The outcome of executing of a data synchronization Job is automatically saved to Nautobot's database for later review.
 * Detailed logging output generated by DiffSync is automatically captured and saved to the database as well.
 
+### Integrations
+
+This plugin supports the following integrations:
```
Comment (chadell) [inline README.md:27]:
```suggestion
This Nautobot application framework includes the following integrations:
```

File: README.md [hunk 17]
```diff
@@ -70,3 +81,23 @@ Any PRs with fixes or improvements are very welcome!
 ## Questions
 
 For any questions or comments, please check the [FAQ](https://docs.nautobot.com/projects/ssot/en/latest/user/faq/) first. Feel free to also swing by the [Network to Code Slack](https://networktocode.slack.com/) (channel `#nautobot`), sign up [here](http://slack.networktocode.com/) if you don't have an account.
+
+## Acknowledgements
+
+This project includes code originally written in separate plugins, which have been merged into this project:
```
Comment (chadell) [inline README.md:87]:
```suggestion
This project includes code originally written in separate Nautobot apps, which have been merged into this project:
```

File: docs/admin/install.md [hunk 18]
```diff
@@ -19,7 +22,13 @@ The plugin is available as a Python package via PyPI and can be installed with `
 pip install nautobot-ssot
 ```
 
-To ensure Single Source of Truth is automatically re-installed during future upgrades, create a file named `local_requirements.txt` (if not already existing) in the Nautobot root directory (alongside `requirements.txt`) and list the `nautobot-ssot` package:
+To use specific integrations, add them as extra dependencies:
+
+```shell
+pip install nautobot-ssot[aci]
```
Comment (chadell) [inline docs/admin/install.md:28]: I would show an example with at least two integrations to show the syntax

File: docs/admin/integrations/aci_setup.md [hunk 19]
```diff
@@ -0,0 +1,137 @@
+# Cisco ACI Integration Setup
+
+This guide will walk you through steps to set up Cisco ACI integration with the `nautobot_ssot` App.
+
+## Prerequisites
+
+Before configuring the integration, please ensure, that `nautobot-ssot` App was [installed with integration extra dependencies](../install.md#install-guide).
```
Comment (chadell) [inline docs/admin/integrations/aci_setup.md:7]:
```suggestion
Before configuring the integration, please ensure, that `nautobot-ssot` App was [installed with the ACI integration extra dependencies](../install.md#install-guide).
```

File: docs/admin/integrations/aci_setup.md [hunk 20]
```diff
@@ -0,0 +1,137 @@
...
+### Per-APIC settings
+
+The APIC URL and credentials need to be created as environment variables on the host system.
+
+You can configure multiple APIC instances for synchronization. To do this, append `_` character, followed by an identifier, to the names of environment variables.
+
+In the example below, configured APIC uses `NTC` for an identifier. Instead of `NTC` you could, for example, use `CHCG01` to configure an APIC instance in your Chicago facility.
+
+```bash
+export NAUTOBOT_APIC_BASE_URI_NTC=https://aci.cloud.networktocode.com
```
Comment (chadell) [inline docs/admin/integrations/aci_setup.md:64]: you could show the other example with devnet that is easier to reproduce

**Review by chadell — COMMENTED** [review]
I like the structure! let's move it forward

File: docs/admin/integrations/aci_setup.md (hunk 20 above)
Comment (snaselj) [reply docs/admin/integrations/aci_setup.md:64]: Copied from the original plugin doc. Could you pleas provide more specific suggestion or ask someone with the devnet account?

**Comment by snaselj** [comment]
Blocked by https://github.com/nautobot/nautobot-plugin-ssot/pull/137

**Comment by snaselj** [comment]
Added all integrations documentation, it's ready for review now.

File: docs/admin/install.md (hunk 18 above)
Comment (snaselj) [reply docs/admin/install.md:29]: Fixed

File: docs/admin/upgrade.md [hunk 21]
```diff
@@ -2,4 +2,28 @@
 
 ## Upgrade Guide
 
-When a new release comes out it may be necessary to run a migration of the database to account for any changes in the data models used by this plugin. Execute the command `nautobot-server post-upgrade` within the runtime environment of your Nautobot installation after updating the `nautobot-ssot` package via `pip`.
+When a new release comes out it may be necessary to run a migration of the database to account for any changes in the data models used by this Nautobot app. Execute the command `nautobot-server post-upgrade` within the runtime environment of your Nautobot installation after updating the `nautobot-ssot` package via `pip`.
+
+### Potential Apps Conflicts
+
+!!! warning
+    If upgrading from `1.x` version to `2.x` version of `nautobot-ssot` app, note that it now incorporates features previously provided by individual apps.
```
Comment (jdrew82) [inline docs/admin/upgrade.md:10]:
```suggestion
    If upgrading from versions prior to 1.4 of the `nautobot-ssot` app, note that it now incorporates features previously provided by individual apps.
```

Comment (jdrew82) [reply docs/admin/upgrade.md:10]: We're going to make 1.4.0 the version for all integrations and 2.x is going to be Nautobot 2.0 compatibility.

File: docs/dev/jobs.md [hunk 22]
```diff
@@ -9,15 +9,12 @@ Familiarity with [DiffSync](https://diffsync.readthedocs.io/en/latest/) and with
...
 3. Define a `DiffSync` adapter class for loading initial data from the Data Source or Data Target system and constructing instances of the `DiffSyncModel` classes to represent that data.
-
-4. Develop a Job class, derived from either the `DataSource` or `DataTarget` classes provided by this plugin, and implement the adapters to populate the `self.source_adapter` and `self.target_adapter` that are used by the built-in implementation of `sync_data`. This `sync_data` method is an opinionated way of running the process including some performance data, more in [next section](#analyze-job-performance), but you could overwrite it completely or any of the key hooks that it calls:
+4. Develop a Job class, derived from either the `DataSource` or `DataTarget` classes provided by this Nautobot app, and implement the adapters to populate the `self.source_adapter` and `self.target_adapter` that are used by the built-in implementation of `sync_data`. This `sync_data` method is an opinionated way of running the process including some performance data, more in [next section](#analyze-job-performance), but you could overwrite it completely or any of the key hooks that it calls:
```
Comment (jdrew82) [inline docs/dev/jobs.md:17]:
```suggestion
4. Develop a Job class, derived from either the `DataSource` or `DataTarget` classes provided by this Nautobot app, and implement the adapters to populate the `self.source_adapter` and `self.target_adapter` methods that are used by the built-in implementation of `sync_data`. This `sync_data` method is an opinionated way of running the process including some performance data, more in [next section](#analyze-job-performance), but you could overwrite it completely or any of the key hooks that it calls:
```

**Review by jdrew82 — CHANGES_REQUESTED** [review]
Just a few tweaks but mostly looking good!

File: docs/user/app_overview.md [hunk 23]
```diff
@@ -1,21 +1,21 @@
 # App Overview
 
-A plugin for Nautobot. This plugin facilitates integration and data synchronization between various "source of truth" (SoT) systems, with Nautobot acting as a central clearinghouse for data - a Single Source of Truth, if you will.
+An app for Nautobot. This Nautobot app facilitates integration and data synchronization between various "source of truth" (SoT) systems, with Nautobot acting as a central clearinghouse for data - a Single Source of Truth, if you will.
 
 !!! note
     Throughout this documentation, the terms "app" and "plugin" will be used interchangeably.
```
[the remark this answers is not in the document]
Comment (snaselj) [reply docs/user/app_overview.md:6]: Tried to replace most `plugin` occurrences, however word `plugin` is still used in package names, and configuration keys. I would keep it here.

File: README.md [hunk 24]
```diff
@@ -46,22 +57,22 @@ More screenshots can be found in the [Using the App](https://docs.nautobot.com/p
...
 - [Administrator Guide](https://docs.nautobot.com/projects/ssot/en/latest/admin/install/) - How to Install, Configure, Upgrade, or Uninstall the App.
-- [Developer Guide](https://docs.nautobot.com/projects/ssot/en/latest/dev/contributing/) - Extending the App, Code Reference, Contribution Guide.
+- [Developer Guide](https://docs.nautobot.com/projects/ssot/en/latest/dev/contributing/) - Extending the App, Developing Jobs, Code Reference, Contribution Guide.
 - [Release Notes / Changelog](https://docs.nautobot.com/projects/ssot/en/latest/admin/release_notes/).
 
 ### Contributing to the Documentation
 
-You can find all the Markdown source for the App documentation under the [`docs`](https://github.com/nautobot/nautobot-plugin-ssot//tree/develop/docs) folder in this repository. For simple edits, a Markdown capable editor is sufficient: clone the repository and edit away.
+You can find all the Markdown source for the app documentation under the [`docs`](https://github.com/nautobot/nautobot-plugin-ssot//tree/develop/docs) folder in this repository. For simple edits, a Markdown capable editor is sufficient: clone the repository and edit away.
```
[the remark this answers is not in the document]
Comment (snaselj) [reply README.md]: Fixed

File: docs/admin/integrations/aristacv_setup.md [hunk 25]
```diff
@@ -0,0 +1,200 @@
...
+| aristacv_verify        | boolean | If False, the integration will download the certificate from CloudVision and trust it for gRPC calls. |
+
+To connect to a cloud instance of CloudVision you must set the following variable:
+
+| Configuration Variable | Type   | Usage                                       | Default           |
+| ---------------------- | ------ | ------------------------------------------- | ----------------- |
+| aristacv_cvaas_url     | string | URL used to connect to your CvaaS instance. | www.arista.io:443 |
+
+When syncing from CloudVis integration will create new Arista devices that do not exist in Nautobot. When creating new devices in Nautobot, a site, device role, device role color, device status, and device are required. You may define which values to use by configuring the following values in your `nautobot_config.py` file. If you define a `default_device_role` and `default_device_status` that already exist, the default color value for both of those will be ignored as it will pull that information from Nautobot.
```
[the remark this answers is not in the document]
Comment (snaselj) [reply docs/admin/integrations/aristacv_setup.md]: Fixed

File: docs/admin/upgrade.md (hunk 21 above)
Comment (snaselj) [reply docs/admin/upgrade.md]: Fixed

File: docs/dev/jobs.md (hunk 22 above)
Comment (snaselj) [reply docs/dev/jobs.md]: Fixed

**Comment by snaselj** [comment]
> Just a few tweaks but mostly looking good!

Thanks, PR is fixed now.

File: docs/admin/integrations/aci_setup.md (hunk 20 above)
Comment (chadell) [reply docs/admin/integrations/aci_setup.md:64]: ok, if you copied the original one, let's keep it as it is

File: docs/user/app_overview.md (hunk 23 above)
[the remark this answers is not in the document]
Comment (jdrew82) [reply docs/user/app_overview.md:6]: Alright, fair enough.

**Comment by jdrew82** [comment]
@snaselj Looks like there's a conflict with the nautobot_config.py file. Can we get that resolved? Looks good otherwise.

**Comment by snaselj** [comment]
Conflicts are resolved now @jdrew82

**Comment by chadell** [comment]
still one conflict with `mkdocs.yml`, @snaselj

**Comment by snaselj** [comment]
@chadell Could you pls. check two different `Code Reference` doc? Probably fine, just to be sure.

https://github.com/nautobot/nautobot-plugin-ssot/pull/152/files#diff-98d0f806abc9af24e6a7c545d3d77e8f9ad57643e27211d7a7b896113e420ed2R115

https://github.com/nautobot/nautobot-plugin-ssot/pull/152/files#diff-98d0f806abc9af24e6a7c545d3d77e8f9ad57643e27211d7a7b896113e420ed2R139

**Comment by chadell** [comment]
> Code Reference

I don't see the code reference doc (`Code Reference: "user/reference.md"`) changing...

Could you point me out on what's different?

**Review by chadell — APPROVED** [review]
LGTM

**Comment by snaselj** [comment]
Merged user code reference into developer and fixed some warnings during mkdocs build.

This PR should be done.

@jdrew82 @chadell

**Review by jdrew82 — APPROVED** [review]
LGTM!

**Pull request merged**